lib/CXGN/Chado/Stock.pm

```diff
@@ -630,20 +630,87 @@ sub get_trials {
     my $geolocation_type_id = SGN::Model::Cvterm->get_cvterm_row($self->get_schema(), 'project location', 'project_property')->cvterm_id();
     my $q;
     if ($stock_type eq 'accession'){
-        $q = "select distinct(project.project_id), project.name, projectprop.value from stock as accession join stock_relationship on (accession.stock_id=stock_relationship.object_id) JOIN stock as plot on (plot.stock_id=stock_relationship.subject_id) JOIN nd_experiment_stock ON (plot.stock_id=nd_experiment_stock.stock_id) JOIN nd_experiment_project USING(nd_experiment_id) JOIN project USING (project_id) FULL OUTER JOIN projectprop ON (project.project_id=projectprop.project_id AND projectprop.type_id=$geolocation_type_id) WHERE accession.stock_id=?;";
-    } else {
-        $q = "select distinct(project.project_id), project.name, projectprop.value from stock JOIN nd_experiment_stock USING(stock_id) JOIN nd_experiment_project USING(nd_experiment_id) JOIN project USING (project_id) FULL OUTER JOIN projectprop ON (project.project_id=projectprop.project_id AND projectprop.type_id=$geolocation_type_id) WHERE stock.stock_id=?;";
+        $q = "SELECT DISTINCT project.project_id, project.name, projectprop.value 
+        FROM stock AS accession 
+        JOIN stock_relationship ON (accession.stock_id=stock_relationship.object_id) 
+        JOIN stock AS plot ON (plot.stock_id=stock_relationship.subject_id) 
+        JOIN nd_experiment_stock ON (plot.stock_id=nd_experiment_stock.stock_id) 
+        JOIN nd_experiment_project USING (nd_experiment_id) 
+        JOIN project USING (project_id) 
+        FULL OUTER JOIN projectprop ON (project.project_id=projectprop.project_id AND projectprop.type_id=$geolocation_type_id) 
+        JOIN cvterm ON (cvterm.cvterm_id=nd_experiment_stock.type_id) 
+        FULL OUTER JOIN breeding_programs ON (breeding_programs.breeding_program_id=project.project_id) 
+        WHERE accession.stock_id=? 
+            AND cvterm.name!='phenotyping_experiment' 
```

Comment:
why is phenotyping_experiment nd_experiment type excluded?

Reply:
Because the join to the nd_experiment_project table produces a bunch of phenotyping nd_experiments that result in each trial ID giving duplicate rows, where one of those rows is a phenotying_experiment and the other row has field_layout (for trials) or analysis_experiment (for stored analyses). Kicking out the phenotyping_experiment cvterm is necessary because if you only kick out analysis_experiments, the row with the phenotyping_experiment cvterm is still there and the query does not exclude analyses.